core/azure-core/src/main/java/com/azure/core/implementation/tracing/Tracer.java

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.tracing;
+
+import com.azure.core.util.Context;
+
+/**
+ * Contract that all tracers must implement to be plug-able into the SDK.
+ */
+public interface Tracer {
+    /**
+     * Creates a new tracing span.
+     *
+     * The {@code context} will be checked for containing information about a parent span. If a parent span is found the
+     * new span will be added as a child, otherwise the span will be created and added to the context and any downstream
+     * start calls will use the created span as the parent.
+     *
+     * @param methodName Name of the method triggering the span creation.
+     * @param context Additional metadata that is passed through the call stack.
+     * @return An updated context object.
+     */
+    Context start(String methodName, Context context);
```

Comment:
Did we end up saying we should not include methodName here, and that it should be hand-written using setAttribute, or will we at the minimum do methodName and allow developers to give more context, a prettier name, etc in the attributes method?

Reply:
What I would suggest is that we introduce a `setName` (or `setDescriptiveName`) method on `Tracer`. It would be the developer of the client libraries responsibility to call this method for each service method. In time we could have linters setup to enforce this, but I wonder if a good fallback would be to essentially keep what we have now - if we get to the start of the span and there is no 'tracing name' information in the Context, that we add the current auto-generated method name by default? In other words, is it better to have *something*, rather than nothing?

Reply:
Agree, I like `setOperationName` or `setSpanName`. falling back to a method name in case nothing was provided is good.

Reply:
Added setSpanName which adds data to the context, when the service call is configuring it's span it will attempt to find the set name and use it or default back to the method call.